lua/lspconfig/jdtls.lua

```diff
@@ -1,61 +1,72 @@
 local configs = require 'lspconfig/configs'
 local util = require 'lspconfig/util'
 local handlers = require 'vim.lsp.handlers'
-local path = util.path
 
 local server_name = 'jdtls'
 
-local cmd = {
-  util.path.join(tostring(vim.fn.getenv 'JAVA_HOME'), '/bin/java'),
-  '-Declipse.application=org.eclipse.jdt.ls.core.id1',
-  '-Dosgi.bundles.defaultStartLevel=4',
-  '-Declipse.product=org.eclipse.jdt.ls.core.product',
-  '-Dlog.protocol=true',
-  '-Dlog.level=ALL',
-  '-Xms1g',
-  '-Xmx2G',
-  '-jar',
-  tostring(vim.fn.getenv 'JAR'),
-  '-configuration',
-  tostring(vim.fn.getenv 'JDTLS_CONFIG'),
-  '-data',
-  tostring(vim.fn.getenv 'WORKSPACE'),
-  '--add-modules=ALL-SYSTEM',
-  '--add-opens java.base/java.util=ALL-UNNAMED',
-  '--add-opens java.base/java.lang=ALL-UNNAMED',
-}
+local function get_java_executable()
+  local executable = vim.fn.getenv('JAVA_HOME') ~= vim.NIL
+    and vim.fn.expand('$JAVA_HOME/bin/java')
+    or  'java'
 
---- The presence of one of these files indicates a root directory.
---
--- We search for configuration files of the most common Java build systems. We
--- could add more build systems, such as Make, but let's keep things simple for
--- now.
-local root_files = {
-  -- Single-module projects
-  {
-    'build.xml', -- Ant
-    'pom.xml', -- Maven
-    'settings.gradle', -- Gradle
-    'settings.gradle.kts', -- Gradle
-  },
-  -- Multi-module projects
-  { 'build.gradle', 'build.gradle.kts' },
-}
+  return vim.fn.has('win32') ~= 0
+    and executable..'.exe'
+    or  executable
+end
+
+local function get_workspace_dir()
+  return vim.fn.getenv('$WORKSPACE') ~= vim.NIL
+    and vim.fn.getenv('WORKSPACE')
+    or  vim.fn.expand('$HOME/workspace')
+end
+
+local function get_jdtls_jar()
+  -- The following case is for legacy cases, should we remove it?
+  if vim.fn.getenv('JAR') ~= vim.NIL then
+    print('Using old $JAR environment variable for JDTLS, refer to documentation and update to JDTLS_HOME')
+    return vim.fn.getenv('JAR')
+  end
+
+  if vim.fn.getenv('JDTLS_HOME') == vim.NIL then
+    error('JDTLS_HOME env variable not set')
+  end
+
+  return vim.fn.expand('$JDTLS_HOME/plugins/org.eclipse.equinox.launcher_*.jar')
+end
+
+local function get_jdtls_config()
+  -- The following case is for legacy cases, should we remove it?
+  if vim.fn.getenv('JDTLS_CONFIG') ~= vim.NIL then
+    print('Using old $JDTLS_CONFIG environment variable for JDTLS, refer to documentation and update to JDTLS_HOME')
+    return vim.fn.getenv('JDTLS_CONFIG')
+  end
 
---- Callback function for the `language/status` notification.
---
--- The server sends a non-standard notification when the status of the language
--- server changes. This can be used to display progress as the server is
--- starting up.
+  if vim.fn.getenv('JDTLS_HOME') == vim.NIL then
+    error('JDTLS_HOME env variable not set')
+  end
+
+  if vim.fn.has('unix') ~= 0 then
```

Comment:
I would use luv's os_uname here instead

Reply:
I couldn't find this function anywhere, where should I look for this?

Reply:
With neovim, you have access to all of luv's functions (bindings to libuv) under vim.loop

https://github.com/luvit/luv/blob/master/docs.md#uvos_uname

vim.loop == uv for those examples

Reply:
You can grep the repo for how we match against windows/other systems